java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture2/tasks/_2/Number2.java

```diff
@@ -0,0 +1,162 @@
+        System.out.println("Your room size is:");
+        System.out.println(robotGame.getRoomSize() + "\n");
+        System.out.println("Your coordinates are:");
+        System.out.println(robotGame.getCoordinates() + "\n");
+        System.out.println("Set direction where your robot is going. Default direction is SOUTH");
+        System.out.println("0. SOUTH,\n1. NORTH,\n2. WEST,\n3. EAST");
+        Direction direction = Direction.values()[Integer.parseInt(scanner.next())];
+        System.out.println(direction.toString() + "\n");
+
+        System.out.println("1.Move Forward. Type N steps:\n2.Change Direction;\n3.Get coordinates;\n4.Get directions;\n5.Exit");
+
+        while (!exit) {
+            switch (scanner.nextInt()) {
+                case 1:
+                    robotGame.moveForward();
+                    System.out.println("Moving On");
+                    break;
+                case 2:
+                    System.out.println("1.Turn Left;\n2.Turn Right;");
+                    if (scanner.nextInt() == 1) {
+                        robotGame.turnLeft();
+                        System.out.println("Turned left");
+
+                    } else {
+                        robotGame.turnRight();
+                        System.out.println("Turned right");
+                    }
+                    break;
+                case 3:
+                    System.out.println(robotGame.getCoordinates());
+                    break;
+                case 4:
+                    robotGame.getDirection();
+                    break;
+                case 5:
+                    System.out.println("Shutting down");
+                    exit = true;
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+}
+
+enum Direction {
+    SOUTH,
+    NORTH,
+    WEST,
+    EAST
+}
+
+class RobotGame {
+
+    private final int width;
+    private final int height;
+    private int x;
+    private int y;
+    Direction direction;
+
+    public RobotGame(int width, int height, int x, int y, Direction direction) {
+        this.width = width;
+        this.height = height;
+        this.x = x;
+        this.y = y;
+        this.direction = direction;
+    }
+
+    public String getRoomSize() {
+        return "Width: " + width + " " + "Height: " + height;
+    }
+
+    public String getCoordinates() {
+        return "X: " + x + " " + "Y: " + y;
+    }
+
+    public void getDirection() {
+        System.out.println(direction);
+    }
+
+    public void setDirection(Direction direction) {
+        this.direction = direction;
+    }
+
+
+    public void moveForward() {
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Type N steps");
+        int n = scanner.nextInt();
+
+        if (direction == Direction.SOUTH) {
+            if (y < height) {
+                y += n;
+            }
+            if (y == height) {
```

Comment:
what if y > height?

e.g., the `height` is `10`, I've entered `n=20`, `y` becomes `20` which is > `10`